### PR TITLE
Mod10

### PR DESCRIPTION
**Pull request by durrellchamorro**

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -20,12 +20,11 @@
...
-    password:             ENV['gmail_password'],
+    address:              ENV['MAILGUN_SMTP_SERVER'],
+    port:                 ENV['MAILGUN_SMTP_PORT'],
+    domain:               "durrellsnetflixclone.herokuapp.com",
+    user_name:            ENV['MAILGUN_SMTP_LOGIN'],
+    password:             ENV['MAILGUN_SMTP_PASSWORD'],
     authentication:       'plain',
     enable_starttls_auto: true  }
-  config.action_mailer.default_url_options = { :host => "durrellsnetflixclone.herokuapp.com" }
 end
```
Comment (durrellchamorro) [inline config/environments/production.rb]: All my tests pass with these mailgun credentials, but I don't understand why in production I'm getting emails from the gmail account I setup before I setup mailgun.

File: app/controllers/invitations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params.merge!(inviter_id: current_user.id))
+    if @invitation.save
+      AppMailer.send_invitation_email(@invitation).deliver
+      flash[:success] = "You sent an invitation to #{@invitation.friends_name}."
+      redirect_to new_invitation_path
+    else
+      flash[:danger] = "You entered incorrect information"
```
Comment (tomtomecek) [inline app/controllers/invitations_controller.rb]: `flash.now` is preferred when you render the template.

File: app/mailers/app_mailer.rb [hunk 3]
```diff
@@ -9,4 +9,9 @@ def send_forgot_password(user)
     @user = user
     mail to: user.email_address, from: "yourjeangenie2@gmail.com", subject: "Please reset your password."
   end
+
+  def send_invitation_email(invitation)
+    @invitation = invitation
+    mail to: invitation.friends_email, from: "yourjeangenie2@gmail.com", subject: "Join MyFlix!"
```
Comment (tomtomecek) [inline app/mailers/app_mailer.rb]: You can refactor `from:` to `default from: ...` above all methods.

File: app/models/invitation.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Invitation < ActiveRecord::Base
+  belongs_to :inviter, class_name: "User"
+  validates_presence_of :friends_name, :friends_email, :invitation_message, :inviter_id
+  before_create :generate_token
```
Comment (tomtomecek) [inline app/models/invitation.rb]: You should generate token upon request, not when user is created. Check notes in the assignment.

File: app/views/invitations/new.html.haml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+            .col-sm-4
+              = f.text_field :friends_name, label: "Friend's Name"
+          .row
+            .col-sm-4
+              = f.email_field :friends_email, label: "Friend's Email Address"
+          .row
+            .col-sm-4
+              = f.text_area :invitation_message, value: "Please join this really cool site!",  size: "50x6"
+        %fieldset.form-group.action
+          %input(type="submit" value="Send Invitation" class="btn btn-default")
```
Comment (tomtomecek) [inline app/views/invitations/new.html.haml]: I would like to see this as `f.submit ...` .

File: config/environments/production.rb (hunk 1 above)
Comment (tomtomecek) [inline config/environments/production.rb]: Heroku has only master branch. `git push heroku master` should be OK. Also check your server ENV vars.

File: spec/features/user_resets_password_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+require 'capybara/email/rspec'
```
Comment (tomtomecek) [inline spec/features/user_resets_password_spec.rb]: If you require it in your spec_helper it will be required here as well.

**Pull request merged**